### PR TITLE
[Merged by Bors] - chore: uniformize `ContDiffWithinAt` API with other local notions

### PR DESCRIPTION
**Pull request by sgouezel**
Currently, some statements are available for `DifferentiableWithinAt` but not for `ContinuousWithinAt` or conversely (and the same for the other `WithinAt` notions), or they are both available with slightly different names or assumptions.

This PR is a first step to uniformize these API: I've gone through the APIs of the different notions, and added all the lemmas that were available for another notion to `ContinuousWithinAt` if they were missing here, and fixed some names. Once this PR is merged, I will open other PRs to make sure that all the other notions also have the same full API.

Notable changes (imported from lemma names of other notions, and that I found better than the `ContinuousWithinAt` version):
* rename `ContinuousWithinAt.insert_self` to `ContinuousWithinAt.insert` 
* rename `continuousWithinAt_congr_nhds` to `continuousWithinAt_congr_set`, and change its assumptions from `𝓝[s] x = 𝓝[t] x` to `s =ᶠ[𝓝 x] t` (see Zulip discussion supporting this change at https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/local.20equality.20of.20sets/near/477653314)
* Added `continuousWithinAt_congr_set'` (taking an assumption `s =ᶠ[𝓝[{y}ᶜ] x] t`)
* Added several variations around `ContinuousWithinAt.congr_of_mem` (both in their implication and iff version)
* Added `of_eq` versions of all the composition lemmas

Also, use `variable` in the `ContinuousOn` file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousOn.lean [hunk 1]
```diff
@@ -777,78 +802,169 @@ theorem Continuous.continuousWithinAt {f : α → β} {s : Set α} {x : α} (h :
...
+theorem ContinuousWithinAt.comp_of_preimage_mem_nhdsWithin_of_eq {g : β → γ} {t : Set β} {y : β}
+    (hg : ContinuousWithinAt g t y) (hf : ContinuousWithinAt f s x) (h : f ⁻¹' t ∈ 𝓝[s] x)
+    (hy : f x = y) :
+    ContinuousWithinAt (g ∘ f) s x := by
+  subst hy; exact hg.comp_of_preimage_mem_nhdsWithin hf h
+
+theorem ContinuousWithinAt.comp_of_mem_nhdsWithin_image {g : β → γ} {t : Set β}
+    (hg : ContinuousWithinAt g t (f x)) (hf : ContinuousWithinAt f s x)
+    (hs : t ∈ 𝓝[f '' s] f x) : ContinuousWithinAt (g ∘ f) s x :=
+  (hg.mono_of_mem_nhdsWithin hs).comp hf (subset_preimage_image f s)
```
Comment (ADedecker) [inline Mathlib/Topology/ContinuousOn.lean:928]:
```suggestion
  (hg.mono_of_mem_nhdsWithin hs).comp hf (mapsTo_image f s)
```
Just for the sake of readability

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Topology/ContinuousOn.lean [hunk 2]
```diff
@@ -675,100 +676,124 @@ theorem continuousOn_isOpen_of_generateFrom {β : Type*} {s : Set α} {T : Set (
 ### Congruence and monotonicity properties with respect to sets
 -/
 
-theorem ContinuousWithinAt.mono {f : α → β} {s t : Set α} {x : α} (h : ContinuousWithinAt f t x)
+theorem ContinuousWithinAt.mono (h : ContinuousWithinAt f t x)
     (hs : s ⊆ t) : ContinuousWithinAt f s x :=
   h.mono_left (nhdsWithin_mono x hs)
 
-theorem ContinuousWithinAt.mono_of_mem {f : α → β} {s t : Set α} {x : α}
```
[the remark this answers is not in the document]
Comment (sgouezel) [reply Mathlib/Topology/ContinuousOn.lean:682]: Some lemmas using the assumption `t ∈ 𝓝[s] x` are named `..._mem`, others are named `..._mem_nhdsWithin`. The latter looks less ambiguous to me (as other lemmas with `of_mem` require that some point belongs to some set), so I chose to uniformize things a little bit.

**Pull request closed without merging**